CoreNotesView/src/au/gov/asd/tac/constellation/views/notes/NotesViewController.java

```diff
@@ -114,7 +115,7 @@ public void read(final GraphReadMethods graph, final PluginInteraction interacti
                 return;
             }
 
-            pane.setNotes(currentState.getNotes());
+            pane.setNotes(new ArrayList<>(currentState.getNotes()));
```

Comment:
Instead of instantiating a new array with the values of currentState.getNotes(), would it possibly be better to have currentState.getNotes() return an unmodifiable list?

Reply:
If going with the new ArrayList is the way to go, are you also able to add a comment explaining why we're creating a new list populated with an existing list

Reply:
Good question @sol695510. If you make it an unmodifiable list you end up with a different exception (i.e `UnsupportedOperationException`) as the code elsewhere is still trying to write to the `List`. It helps confirm the issue of something reading and writing at the same time but does not rectify the root cause which is while the List is being read, something else is changing the list.

My understanding is `Collections.unmodifieble()` is used more when you want give the list to someone you don't want them to change. In this case, changing the `getNotes()` to `return Collections.unmodifiableList(notesViewEntries)` could be something we do as a protection layer and indicates to anyone trying to use `getNotes()` that they can only read from it.

I did a test with this and removed the `new ArrayList` and got the following error, to help confirm what I meant above, i.e it won't solve the root cause.

```
java.lang.UnsupportedOperationException
	at java.base/java.util.Collections$UnmodifiableCollection.clear(Collections.java:1079)
	at au.gov.asd.tac.constellation.views.notes.state.NotesViewState.setNotes(NotesViewState.java:65)
	at au.gov.asd.tac.constellation.views.notes.NotesViewController$NotesViewStateWriter.edit(NotesViewController.java:153)
	at au.gov.asd.tac.constellation.plugins.templates.SimpleEditPlugin.describedEdit(SimpleEditPlugin.java:207)
[catch] at au.gov.asd.tac.constellation.plugins.templates.SimpleEditPlugin.run(SimpleEditPlugin.java:115)
Caused: java.lang.RuntimeException
	at au.gov.asd.tac.constellation.plugins.templates.SimpleEditPlugin.run(SimpleEditPlugin.java:135)
[catch] at au.gov.asd.tac.constellation.graph.node.plugins.DefaultPluginEnvironment.lambda$executePluginLater$0(DefaultPluginEnvironment.java:129)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:834)
```

I could be wrong but the underlying issue is that there are many reads and writes in quick succession which is happening on any graph change (because it triggers a plugin which then causes the state to update) and if that could be fixed I feel the concurrent mod would be a rare thing to occur. We could then potentially do away with the `new ArrayList` if its unlikely to be reproduced (but it is still an issue).

Another solution to all of this is to use `synchronized` and lock the reads and writes but that would slow the entire app down because everything would have to wait for the read and write states which is happening all the time.

Does that help clarify things?

Reply:
@antares1470 I've updated this with a comment as suggested.